hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterClockImpl.java

```diff
@@ -46,4 +47,16 @@ public void setMasterTime(long masterTime) {
     public long getClusterTimeDiff() {
         return (clusterTimeDiff == Long.MAX_VALUE) ? 0 : clusterTimeDiff;
     }
+
+    @Override
+    public long getClusterStartTime() {
```

Comment:
@ahmetmircik , You are totally right it is fixed with java 8. However, the fix they made requires some uptime counter which monotonically increases and it is implemented with native code.(You can see it here :  https://bugs.openjdk.java.net/browse/JDK-8027294) 

In java version 8, RuntimeMXBean uses value of the monotonic counter.(http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8-b132/sun/management/VMManagementImpl.java#183) However versions before 8, they are subtracting System.currentTimeMillis() value from vmStartTime value which causes the negative values for uptime of the jvm. (http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/7-b147/sun/management/RuntimeImpl.java#118). So the perfect solution for this case, would require a cluster wide monotonic counter. Apparently, it took 18 years for JVM guys to write a uptime counter for a single JVM :), writing it for multiple jvm clusters would be challenging task.

I am using System.nanoTime() for now, because java doc says : "This method can only be used to measure elapsed time and is not related to any other notion of system or wall-clock time."  It seems to be not affected by system clock changes and preserves its monotonicity only if the underlying OS does. There is bug ticket for such case (https://bugs.openjdk.java.net/browse/JDK-6458294).  At least using nano time seems to solve negative values for uptime in most OS.

I also updated java doc and stated that ClusterClock.getClusterStartTime() should be used with care and it is only used for statistical purposes.